pallet-chainlink-feed/src/lib.rs

```diff
@@ -862,7 +872,22 @@ pub mod pallet {
 					Details::<T>::insert(feed_id, round_id, details);
 				}
 
-				Ok(().into())
+				Ok((
+					None,
+					match T::SubmitterPaysFee::get() {
+						SubmitterPaysFee::Always => Pays::Yes,
+						SubmitterPaysFee::FreeForValidRound => <Feed<T>>::load_from(feed_id)
+							.map(|feed| {
+								if feed.ensure_valid_round(&oracle, round_id).is_ok() {
+									Pays::No
+								} else {
+									Pays::Yes
+								}
+							})
+							.unwrap_or(Pays::Yes),
+					},
```

Comment:
Is it this necessary to check everything again here?
That's the end of the extrinsic, everything succeeded, so from my understanding, at this point we only the to check what kind of `SubmitterPaysFee` we have.
This would also mean that we can remove the `ensure_valid_round` function entirely, right?

Reply:
https://github.com/ChainSafe/chainlink-polkadot/pull/13/files#r630374518

Discussed this before, `ensure_valid_round` is a different check out of the other logic in this function, and if we remove `ensure_valid_round` entirely, our `SubmitterPaysFee` should be like

```rust
enum SubmitterPaysFee {
    Yes,
    No,
}
```

Reply:
Ah, sorry I missed:

https://github.com/ChainSafe/chainlink-polkadot/pull/25/files#diff-987a170684786ff2b29a8acc32e56cacfe49e9158322b26e36cc67ec6a20c226L755

can't we keep this line as it is and then convert `SubmitterPaysFee` into `Pays` at the end?
Because `ensure_valid_round` only validates the round and nothing else, like whether the value is valid. These checks follow. So at the end of the extrinsic we can make it `Pays::No` if `SubmitterPaysFee::FreeForValidRoud`.

I see now how the naming might be confusing, should we rename `SubmitterPaysFee::FreeForValidRoud` to `SubmitterPaysFee::FreeForValidSubmission` instead?